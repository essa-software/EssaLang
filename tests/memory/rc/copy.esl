struct Foo {
    // Make it non-copyable
    func __drop__(this) {
        print("Foo dropped: {}\n", this.field);
    }
    field: u32;
}

func main() {
    // FIXME: This is not a best test, but I don't think that a more
    // complicated situation is representable in ESL yet.
    let foo = rc(Foo { field: 32 }); // &mut Foo
    {
        let foo_ref = foo; // another &mut Foo
    }
    print("foo is still alive\n");
}
