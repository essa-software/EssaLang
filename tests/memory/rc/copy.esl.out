foo is still alive
Foo dropped: 32
